pkg/controller/gkenetworkparamset/gkenetworkparamset_controller.go

```diff
@@ -311,9 +312,26 @@ func (c *Controller) syncNetworkWithGNP(ctx context.Context, network *networkv1.
 	return nil
 }
 
-func (c *Controller) handleGKENetworkParamSetDelete(ctx context.Context, params *networkv1.GKENetworkParamSet) error {
+func (c *Controller) handleGKENetworkParamSetDelete(ctx context.Context, params *networkv1.GKENetworkParamSet, key string) error {
+	if params.Status.NetworkName == "" {
+		removeFinalizerInPlace(params)
+		return nil
+	}
+
+	network, err := c.networkClientset.NetworkingV1().Networks().Get(ctx, params.Status.NetworkName, v1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			removeFinalizerInPlace(params)
+			return nil
+		}
+		return err
+	}
 
-	removeFinalizerInPlace(params)
+	if network.InUse() {
+		c.queue.AddAfter(key, 1*time.Minute) // recheck network in use again later
```

Comment:
but if you return an error it will requeued too, 

what is the motivation to enqueue, you should document this decisions with comments

Reply:
I agree with Antonio, shouldn't we just return an error here? Why are we picking 1 minute?

Reply:
errors are only requeued 5 times before being dropped by the queue.

Manually adding a check after a minute will be done indefinitely so we keep checking if the network is in use.

Reply:
I figured this wasnt an issue, but youre right, we can listen for the network update.

Reply:
well, image that you have 100 networks that don't change, do you keep spinning around every 1 minutes each of them?

Reply:
fixed.